server/asset/eth/rpcclient.go

```diff
@@ -120,3 +120,13 @@ func (c *rpcclient) swap(ctx context.Context, secretHash [32]byte) (*swap.ETHSwa
 func (c *rpcclient) transaction(ctx context.Context, hash common.Hash) (tx *types.Transaction, isMempool bool, err error) {
 	return c.ec.TransactionByHash(ctx, hash)
 }
+
+// accountBalance gets the account balance, including the effects of known
+// unmined transactions.
+func (c *rpcclient) accountBalance(ctx context.Context, addr common.Address) (*big.Int, error) {
+	bigBal, err := c.ec.PendingBalanceAt(ctx, addr)
```

Comment:
I wonder how this works with multiple txns with the same nonce, e.g. replacements commonly used to either cancel a txn or speed up a txn

Reply:
Exactly, a 0 ETH transaction to yourself.  At least that's how I've seen it on etherscan.